Content.Shared/DoAfter/SharedDoAfterSystem.Update.cs

```diff
@@ -164,16 +162,24 @@ private bool ShouldCancel(DoAfter doAfter,
         if (args.Target is { } target && !xformQuery.TryGetComponent(target, out targetXform))
             return true;
 
-        TransformComponent? usedXform = null;
-        if (args.Used is { } @using && !xformQuery.TryGetComponent(@using, out usedXform))
-            return true;
-
         // TODO: Re-use existing xform query for these calculations.
-        if (args.BreakOnMove && !(!args.BreakOnWeightlessMove && _gravity.IsWeightless(args.User, xform: userXform)))
+        if (args.BreakOnMove)
         {
-            // Whether the user has moved too much from their original position.
-            if (!_transform.InRange(userXform.Coordinates, doAfter.UserPosition, args.MovementThreshold))
-                return true;
+            if (!(!args.BreakOnWeightlessMove && _gravity.IsWeightless(args.User, xform: userXform)))
```

Comment:
```suggestion
            // Check the distance moved if the user is in gravity or if the doafter breaks on distance in space.
            if (!_gravity.IsWeightless(args.User, xform: userXform) || args.BreakOnWeightlessMove)
```

Reply:
I know the original code has a double `!` but I don't think that's necessary. Just makes the code harder to understand.